constantine/ethereum_eip7594_kzg.nim

```diff
@@ -0,0 +1,196 @@
+import
+  std/typetraits,
+
+  constantine/named/algebras,
+  ./math/io/[io_bigints, io_fields],
+  ./math/[ec_shortweierstrass, arithmetic, extension_fields],
+  ./math/arithmetic/limbs_montgomery,
+  ./math/polynomials/polynomials,
+  ./math/arithmetic/bigints,
+  ./commitments/kzg,
+  ./hashes,
+  ./platforms/[abstractions, allocs],
+  ./serialization/[codecs_status_codes, codecs_bls12_381, endians],
+  ./commitments_setups/ethereum_kzg_srs
+  # ./ethereum_eip4844_kzg
+
+export trusted_setup_load, trusted_setup_delete, TrustedSetupFormat, TrustedSetupStatus, EthereumKZGContext
+
+const RANDOM_CHALLENGE_KZG_CELL_BATCH_DOMAIN=asBytes"RCKZGCBATCH__V1_"
+const FIELD_ELEMENTS_PER_BLOB=128
+const FIELD_ELEMENTS_PER_CELL=64
+const BYTES_PER_FIELD_ELEMENT=64
+
+# Derived
+# ------------------------------------------------------------
+const BYTES_PER_CELL= FIELD_ELEMENTS_PER_CELL*BYTES_PER_FIELD_ELEMENT
+const FIELD_ELEMENTS_PER_EXT_BLOB=2*FIELD_ELEMENTS_PER_BLOB
+
+# The number of cells in an extended blob
+const CELLS_PER_EXT_BLOB=int(FIELD_ELEMENTS_PER_EXT_BLOB/FIELD_ELEMENTS_PER_CELL)
+
+
+type 
+  # A polynomial in coefficient form
+  PolyCoeff = PolynomialCoef[FIELD_ELEMENTS_PER_EXT_BLOB,Fr[BLS12_381]]
+  # The evaluation domain of a cell
+  Coset* = distinct array[FIELD_ELEMENTS_PER_CELL, Fr[BLS12_381]] 
+  # The internal representation of a cell (the evaluations over its Coset)
+  CosetEvals* = distinct array[FIELD_ELEMENTS_PER_CELL, Fr[BLS12_381]]
+  # The unit of blob data that can come with its own KZG proof
+  Cell* = array[BYTES_PER_CELL, byte]
+  # Validation: x < CELLS_PER_EXT_BLOB
+  CellIndex* = uint64 
```

Comment:
Is there a need to export all of those?

For testing `import {.all.}` can be used.

Note that if you get too much boilerplates with distinct types in the future, for example when you want to use array accesses `[]`, you can use an object instead:

```Nim
type Coset = object
  raw: array[FIELD_ELEMENTS_PER_CELL, Fr[BLS12_381]]
```

Reply:
Was referencing to eip4884 while writing this, should I change this still? @mratsim

Reply:
Will the Cell and CellIndex be used from other files/modules except tests? If not they can be private.